### PR TITLE
fix nat64 well-known prefix check

### PR DESCRIPTION
**Pull request by sukunrt**

### CODE REVIEW

File: net/ip.go [hunk 1]
```diff
@@ -118,13 +117,20 @@ func zoneless(m ma.Multiaddr) ma.Multiaddr {
 	}
 }
 
-var NAT64WellKnownPrefix = [4]byte{0x0, 0x64, 0xff, 0x9b}
+var NAT64WellKnownPrefix *net.IPNet
```
[the remark this answers is not in the document]
Comment (sukunrt) [reply net/ip.go:120]: maybe? This satisfies *net.IPNet satisfies `net.ParseCIDR` https://pkg.go.dev/net#ParseCIDR

[the remark this answers is not in the document]
Comment (marten-seemann) [reply net/ip.go:120]:
```go
_, net, err := net.ParseCIDR("64:ff9b::/96")
// error handling
NAT64WellKnownPrefix = *net
```

[the remark this answers is not in the document]
Comment (sukunrt) [reply net/ip.go]: I've changed it, but why do we prefer the struct?

File: net/ip.go [hunk 2]
```diff
@@ -118,13 +117,21 @@ func zoneless(m ma.Multiaddr) ma.Multiaddr {
...
 }
 
-var NAT64WellKnownPrefix = [4]byte{0x0, 0x64, 0xff, 0x9b}
+var NAT64WellKnownPrefix net.IPNet
 
-// IsNAT64IPv4ConvertedIPv6Addr returns whether addr is an IPv6 address that begins with
-// the well-known prefix "64:ff9b" used for NAT64 Translation
-// see RFC 6052
+func init() {
+	var err error
```
Comment (marten-seemann) [inline net/ip.go:123]:
```suggestion
```

File: net/ip.go [hunk 3]
```diff
@@ -118,13 +117,21 @@ func zoneless(m ma.Multiaddr) ma.Multiaddr {
 	}
 }
 
-var NAT64WellKnownPrefix = [4]byte{0x0, 0x64, 0xff, 0x9b}
+var NAT64WellKnownPrefix net.IPNet
```
Comment (marten-seemann) [inline net/ip.go:120]: This doesn't need to be exposed, right?

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**